lib/filewriter.js

```diff
@@ -0,0 +1,182 @@
+        var mode = this.isFirstOpen ? this.openmode : this.reopenmode;
+        // setTimeout would be faster than Date.now, but must guarantee .05 sec
+        this.reopenTime = Date.now() + this.reopenInterval;
+        fs.open(this.filename, mode, function(err, fd) {
+            if (err) {
+                if (err.message.indexOf("ENOENT, ") === 0 &&
+                    self.openmode[0] !== 'r' &&
+                    !self.isFirstOpen)
+                {
+                    // if file was removed since the last open, and ok to create, create it anew
+                    self.isFirstOpen = true;
+                    return self._reopenFd(callback);
+                }
+                return callback(err);
+            }
+            self.isFirstOpen = false;
+            self.fd = fd;
+            callback(null, fd);
+        });
+    };
+
+    this._getFd = function(callback) {
+        if (this.fd !== undefined) return callback(null, this.fd);
+        else return this._reopenFd(callback);
+    };
+
+    this._getLockedFd = function(callback) {
+        var self = this;
+        this._getFd(function(err, fd) {
+            if (err) return callback(err);
+            flock(fd, "ex", function(err) {
+                if (err) return callback(err);
+                if (Date.now() < self.reopenTime) {
+                    // grab and lock the fd and only then test reopenTime, since
+                    // the expectation is that most writes will be bunched
+                    // NOTE: this is also needed for correct renameFile
+                    return callback(null, fd);
+                }
+                else {
+                    flockSync(fd, "un");
+                    self._reopenFd(function(err, fd) {
+                        if (err) return callback(err);
+                        self._getLockedFd(callback);
+                    });
+                }
+            });
+        })
+    };
+
+    // atomically append the string to the file
+    this.write = function(str, cb) {
+        var self = this;
+        this._getLockedFd(function(err, fd) {
+            if (err) return cb(err);
+            // not much benefit to reusing a buffer for the writes
+            var buf = new Buffer(str);
+            fs.write(fd, buf, 0, buf.length, null, function(err, nb) {
+                flockSync(fd, "un");
+                cb(err, nb);
+            });
+        });
+    };
+
+    // close the file
+    this.close = function( ) {
+        try { if (this.fd) fs.closeSync(this.fd); }
+        catch (err) { }
+        this.fd = undefined;
+    };
+
+    // expose class method on instance
+    this.mutexTimeout = FileWriter.mutexTimeout;
+    this.renameFile = FileWriter.renameFile;
+}
+
+
+// rename the file and wait for write activity to cease
+FileWriter.mutexTimeout = 5000;
+FileWriter.renameFile = function renameFile( oldName, newName, waitMs, cb ) {
+    if (!cb && typeof waitMs === 'function') {
+        cb = waitMs
```

Comment:
sure.  This too is a historical artifact -- some months ago I switched from semicolon-ful
javascript to semicolon-less javascript as exemplfied by [mongolian](https://npmjs.com/package/mongolian) and [through](https://npmjs.com/package/through), and the renameFile function
was written in the non-semicolon style.